code/game/machinery/slotmachine.dm

```diff
@@ -1,63 +1,71 @@
-/obj/machinery/slot_machine
+/obj/machinery/economy/slot_machine
 	name = "slot machine"
 	desc = "Gambling for the antisocial."
 	icon = 'icons/obj/economy.dmi'
 	icon_state = "slots-off"
 	anchored = TRUE
 	density = TRUE
+
 	var/plays = 0
 	var/working = FALSE
-	var/datum/money_account/account = null
-	var/result = null
-	var/resultlvl = null
+	var/result
+	var/resultlvl
+
+	var/datum/money_account/user_account
 
-/obj/machinery/slot_machine/attack_hand(mob/user as mob)
+/obj/machinery/economy/slot_machine/Initialize(mapload)
+	. = ..()
+	reconnect_database()
+
+/obj/machinery/economy/slot_machine/attack_hand(mob/user as mob)
 	ui_interact(user)
 
-/obj/machinery/slot_machine/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+/obj/machinery/economy/slot_machine/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
 	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
 		ui = new(user, src, ui_key, "SlotMachine", name, 350, 200, master_ui, state)
 		ui.open()
 
-/obj/machinery/slot_machine/ui_data(mob/user)
+/obj/machinery/economy/slot_machine/ui_data(mob/user)
 	var/list/data = list()
 	// Get account
-	account = user.get_worn_id_account()
-	if(!account)
-		if(istype(user.get_active_hand(), /obj/item/card/id))
-			account = get_card_account(user.get_active_hand())
+	user_account = user.get_worn_id_account()
+	if(!user_account)
+		var/obj/item/card/id/id_card = user.get_active_hand()
+		if(istype(id_card))
+			user_account = id_card.get_card_account()
 		else
-			account = null
+			user_account = null
 
 
 	// Send data
 	data["working"] = working
-	data["money"] = account ? account.money : null
+	data["money"] = user_account?.credit_balance
 	data["plays"] = plays
 	data["result"] = result
 	data["resultlvl"] = resultlvl
 	return data
 
-/obj/machinery/slot_machine/ui_act(action, params)
+/obj/machinery/economy/slot_machine/ui_act(action, params)
 	if(..())
 		return
 	add_fingerprint(usr)
 
 	if(action == "spin")
 		if(working)
 			return
-		if(!account || account.money < 50)
-			return
-		if(!account.charge(50, null, "Bet", "Slot Machine", "Slot Machine"))
+		if(!user_account || user_account.credit_balance < 50)
```

Comment:
Forgot to update new slot machine cost here, and i think you might want to tweak the reward numbers (like the lowest one being the same amount as the cost to play the slot machine).

Reply:
fixed in next commit